src/blocked.c

```diff
@@ -564,7 +564,10 @@ static void handleClientsBlockedOnKey(readyList *rl) {
         listIter li;
         listRewind(clients,&li);
 
-        while((ln = listNext(&li))) {
+        /* Avoid processing more than the initial count so that we're not stuck
+         * in an endless loop in case the reprocessing of the command blocks again. */
+        long count = listLength(clients);
+        while ((ln = listNext(&li)) && count--) {
```

Comment:
i don't think we can... in this case we're requesting messages not delivered to anyone else.
so unlike the `$`, which you can replace with a specific ID that's present at the time of first processing, in case of `>` it must remain `>`.

I do think that this of breaking that loop is the right one, to make sure it doesn't process anything that wasn't present in list when we began processing it.
i'm just not very happy with the count based approach, and would have been happier to somehow stash the list on the stack, place an empty one in the server struct, and only process the one on the stack.
but that seems to be complicated.

Reply:
I agree @oranagra - can you say what is the specific concern with the count approach? is it that it reshuffle the list?

Reply:
not sure, it's just less clean than the other one.
for instance, maybe somehow a single client removes 2 elements when processed, and then the counting approach will process one element that we didn't intend for.
not really an important concern, just wonder if we can find something cleaner.